File: java/Interpreter/src/AddExpression.java
// Compound expression
public class AddExpression implements Expression
{
    Expression leftExpression;
    Expression rightExpression;

    public AddExpression(Expression left, Expression right)
    {
        leftExpression = left;
        rightExpression = right;
    }
    
    // interpret each number expression and add them
	public int interpret() {
		return leftExpression.interpret() + rightExpression.interpret();
	}
}

File: ruby/Interpreter/add_expression.rb
require_relative 'expression'

# compound expression
class AddExpression < Expression
  attr_accessor :leftExpression
  attr_accessor :rightExpression

  def initialize(left, right)
    @leftExpression = left
    @rightExpression = right
  end

  # interpret each number and add them
  def interpret
    return @leftExpression.interpret + @rightExpression.interpret
  end
end